presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp

```diff
@@ -146,80 +315,45 @@ class PartitionAndSerializeOperator : public Operator {
   // Replicates the first value to all the empty partitions.
   vector_size_t replicateAnyToEmptyPartitions(
       FlatVector<int32_t>& partitionsVector,
-      FlatVector<StringView>& dataVector,
-      vector_size_t nextRow) {
+      FlatVector<StringView>& dataVector) {
     auto* rawPartitions = partitionsVector.mutableRawValues();
-    const auto numInput = input_->size();
-    std::set<int32_t> nonEmptyPartitions(
-        rawPartitions, rawPartitions + numInput);
-    StringView anyValue = dataVector.valueAt(0);
-
-    int32_t outputRow = nextRow;
-    for (auto idx = 0; idx < numPartitions_; ++idx) {
-      // TODO: Optimize so that we don't need the lookup each partitionId
-      if (nonEmptyPartitions.find(idx) == nonEmptyPartitions.end()) {
-        rawPartitions[outputRow] = idx;
-        dataVector.setNoCopy(outputRow, anyValue);
-        ++outputRow;
+
+    vector_size_t index = 0;
+    for (auto i = 0; i < numPartitions_; ++i) {
+      if (!nonEmptyPartitions.contains(i)) {
+        rawPartitions[index] = i;
+        dataVector.setNoCopy(index, replicateAnyData);
```

Comment:
Don't we also need to add a string buffer to dataVector?

Reply:
We already add string buffer to dataVector while creating outputBuffer. see`createOutputRowVector`.